lib/time_extensions.rb

```diff
@@ -0,0 +1,16 @@
+module TimeExtensions
+  def min_of_day = hour * 60 + min
+  def prev_min = self - 1.minute
+end
```

Comment:
These methods helped cleanup the code quite a bit. Let me know if this is something you think is a good idea.